packages/muelu/adapters/tpetra/MueLu_TpetraOperator_decl.hpp

```diff
@@ -98,22 +101,25 @@ namespace MueLu {
     //! Indicates whether this operator supports applying the adjoint operator.
     bool hasTransposeApply() const;
 
+    //! Clone
     template <class NewNode>
     Teuchos::RCP< TpetraOperator<Scalar, LocalOrdinal, GlobalOrdinal, NewNode> >
-    clone(const RCP<NewNode>& new_node) const {
-      return Teuchos::rcp (new TpetraOperator<Scalar, LocalOrdinal, GlobalOrdinal, NewNode> (Hierarchy_->template clone<NewNode> (new_node)));
-    }
+    clone(const RCP<NewNode>& new_node) const;
```

Comment:
You can't actually hide the implementation of `clone` in the `_def` header, unless you do ETI in a complicated way over all enabled Node types not equal to the current Node type.  Tpetra would actually love to get rid of `clone` at some point -- maybe you could just make this a stub implementation, as it's not clear that we need `clone` any more.

Reply:
Time to end the Attack of the Clones?

Reply:
https://www.youtube.com/watch?v=sLAg9k79tNc

Reply:
Note that I am change the clone interfaces in Tpetra to remove use of Node in the argument list.
This removal will likely propagate to Xpetra and MueLu (sigh).

Reply:
They're there in Xpetra/MueLu because they have to be for inheritance.  I doubt they're used and/or well tested.

Reply:
@csiefer2 @kddevin Maybe now _is_ a good time to remove `clone`?
1. We'll need to touch all the `clone` methods anyway.
2. Test coverage for `clone` is poor.  Some implementations may not even compile.

Reply:
There's an Order 66 joke here somewhere...

Reply:
Order 66 jokes are like dead baby jokes ;-P

Reply:
As per @etphipp we can shoot all of the clone stuff.  He was the guy who needed it in the first place, and he doesn't any more.

(FYI @jhux2 )

Reply:
See #4900